remotes/docker/registry.go

```diff
@@ -156,8 +183,32 @@ func ConfigureDefaultRegistries(ropts ...RegistryOpt) RegistryHosts {
 			Capabilities: HostCapabilityPull | HostCapabilityResolve | HostCapabilityPush,
 		}
 
-		if config.Client == nil {
-			config.Client = http.DefaultClient
+		if opts.client == nil && opts.certRootDir != "" {
```

Comment:
Can we support non-nil `opts.client`?

Reply:
yes

Reply:
Hi, I'm worry about scan certsdir will make pull image rate reduce, does it? and there are any other suggestion?